stack-graphs/src/stitching.rs

```diff
@@ -736,15 +761,21 @@ impl<'a> Display for DisplaySymbolStackKey<'a> {
 /// [`find_all_complete_partial_paths`]: #method.find_all_complete_partial_paths
 pub struct ForwardPartialPathStitcher<H> {
     candidates: Vec<H>,
-    queue: VecDeque<(PartialPath, AppendingCycleDetector<H>)>,
+    extensions: Vec<(PartialPath, AppendingCycleDetector<H>)>,
+    queue: VecDeque<(PartialPath, AppendingCycleDetector<H>, bool)>,
     // tracks the number of initial paths in the queue because we do not want call
     // extend_until on those
     initial_paths: usize,
     // next_iteration is a tuple of queues instead of an queue of tuples so that the path queue
     // can be cheaply exposed through the C API as a continuous memory block
-    next_iteration: (VecDeque<PartialPath>, VecDeque<AppendingCycleDetector<H>>),
+    next_iteration: (
+        VecDeque<PartialPath>,
+        VecDeque<AppendingCycleDetector<H>>,
+        VecDeque<bool>,
+    ),
```

Comment:
In the future, if this grows another field, we'll probably want to turn this into a named struct with named fields instead of a tuple

Reply:
Probably a good idea, yes :) It's only internal, so it doesn't pollute the API, but it becoming a bit unwieldy like this :P